src/main/scala/operations/backfill/MilvusBackfill.scala

```diff
@@ -0,0 +1,569 @@
+          databaseName = config.databaseName
+        )
+      )
+
+      val pkName = client.getPKName(config.databaseName, config.collectionName) match {
+        case scala.util.Success(name) => name
+        case scala.util.Failure(e) =>
+          return Left(ConnectionError(
+            message = s"Failed to get primary key name for collection ${config.collectionName}: ${e.getMessage}",
+            cause = Some(e)
+          ))
+      }
+
+      // Find the primary key field in original data
+      val pkField = originalDF.schema.fields.find(_.name == pkName)
+        .getOrElse {
+          return Left(SchemaValidationError(
+            s"Original collection data must have primary key field '$pkName'"
+          ))
+        }
+
+      // Find the pk field in new field data
+      val newPkField = backfillDF.schema.fields.find(_.name == "pk")
+        .getOrElse {
+          return Left(SchemaValidationError("New field data must have 'pk' field"))
+        }
+
+      // Validate types match
+      if (pkField.dataType != newPkField.dataType) {
+        return Left(SchemaValidationError(
+          s"Primary key type mismatch: original=${pkField.dataType}, new=${newPkField.dataType}"
+        ))
+      }
+
+      Right(pkName)
+
+    } catch {
+      case e: Exception =>
+        logger.error("Failed to validate schema compatibility", e)
+        Left(ConnectionError(
+          message = s"Failed to validate schema compatibility: ${e.getMessage}",
+          cause = Some(e)
+        ))
+    } finally {
+      if (client != null) {
+        try {
+          client.close()
+        } catch {
+          case _: Exception => // Ignore close errors
+        }
+      }
+    }
+  }
+
+  /**
+   * Perform left join between original and new field data
+   */
+  private def performJoin(
+      originalDF: DataFrame,
+      backfillDF: DataFrame,
+      pkName: String
+  ): DataFrame = {
+    originalDF.join(backfillDF, originalDF(pkName) === backfillDF("pk"), "left")
+  }
+
+  /**
+   * Retrieve Milvus metadata (collection ID, partition ID)
+   */
+  private def retrieveMilvusMetadata(
+      config: BackfillConfig
+  ): Either[BackfillError, (Long, Long)] = {
+    var client: MilvusClient = null
+    try {
+      client = MilvusClient(
+        MilvusConnectionParams(
+          uri = config.milvusUri,
+          token = config.milvusToken,
+          databaseName = config.databaseName
+        )
+      )
+
+      val segments = client.getSegments(config.databaseName, config.collectionName)
+        .getOrElse {
+          return Left(ConnectionError(
+            message = s"No segments found for collection ${config.collectionName}"
+          ))
+        }
+
+      if (segments.isEmpty) {
+        return Left(ConnectionError(
+          message = s"Collection ${config.collectionName} has no segments"
+        ))
+      }
+
+      val firstSegment = segments.head
+      Right((firstSegment.collectionID, firstSegment.partitionID))
```

Comment:
If we have many partitions for a collection, will returning the first seg's partitionID be a problem?